src/packagers/JSConcatPackager.js

```diff
@@ -54,13 +54,27 @@ class JSConcatPackager extends Packager {
 
       js = js.split(depName).join(moduleName);
 
-      js = js
-        // $[asset.id]$named_import$a_js => $[module.id]$named_exports
-        .split('$' + asset.id + '$named_import$' + t.toIdentifier(dep.name))
-        .join('$' + mod.id + '$named_export')
-        // $[asset.id]$expand_exports$a_js => $parcel$expand_exports([module.id])
-        .split('$' + asset.id + '$expand_exports$' + t.toIdentifier(dep.name))
-        .join('$parcel$expand_exports(' + mod.id + ',' + asset.id + ')');
+
+      if (dep.isES6Import) {
```

Comment:
We track imports to see if they are from ES6, as well as exports to mark a module as ES6. If they are both ES6, then we can just replace the import names with exports. If a commonjs module is imported with ES6, then the `default` import resolves to `module.exports`, and other names resolve to `module.exports.NAME`.